src/porepy/examples/geothermal_reservoir.py

```diff
@@ -0,0 +1,493 @@
+                    self.time_manager.time,
+                ),
+                "Pa",
+            )
+        return values
+
+    def bc_values_temperature(self, bg: pp.BoundaryGrid) -> np.ndarray:
+        """Return boundary values for temperature on all boundaries.
+
+        Parameters:
+            sd: The subdomain for which to return the BC values.
+
+        Returns:
+            The boundary values for temperature on the given subdomain.
+        """
+        sd = bg.parent
+        values = super().bc_values_temperature(bg)  # type: ignore[misc]
+        if self.is_well_grid(sd):
+            # Retrieve well protocol.
+            well_tag = self.well_names[sd.tags["parent_well_index"]]
+            protocol = self.well_protocols()[well_tag]
+            # Find indices of the well boundary sides.
+            domain_sides = self.domain_boundary_sides(bg)
+            inds = domain_sides.top if self.nd == 3 else domain_sides.north
+            # Set temperature values according to the well protocol.
+            values[inds] = self.units.convert_units(
+                self.get_well_value(
+                    protocol["temperatures"],
+                    self.time_manager.schedule,
+                    self.time_manager.time,
+                ),
+                "K",
+            )
+        return values
+
+    def is_well_grid(self, sd: pp.Grid) -> bool:
+        """Check if a given subdomain grid is a well grid.
+
+        Parameters:
+            sd: The subdomain grid to check.
+
+        Returns:
+            True if the subdomain grid is a well grid, False otherwise.
+        """
+        return "parent_well_index" in sd.tags
+
+    def get_well_value(
+        self,
+        values: np.ndarray,
+        times: np.ndarray,
+        current_time: float,
+    ) -> float:
+        """Return the well value at the current time.
+
+        Parameters:
+            values: Array with time-dependent well values.
+            times: Array with time points corresponding to the well values.
+                Linear interpolation is used to find the value at the current time.
+            current_time: Current simulation time.
+
+        Returns:
+            The well value at the current time.
+        """
+        if current_time < times[0]:
+            raise ValueError("Current time is before the start of the well protocol.")
+        elif current_time > times[-1]:
+            raise ValueError("Current time is after the end of the well protocol.")
+        else:
+            return float(np.interp(current_time, times, values))
+
+    def well_protocols(self) -> dict[str, dict[str, NDArray[np.float64]]]:
+        """Dictionary mapping well tags to well protocols.
+
+        Returns:
+            Dictionary with well protocols, each containing a dictionary with
+            time-dependent temperatures and pressures, with each value being an array of
+            size equal to the number of time steps.
```

Comment:
Scheduled time steps? By default, time steps are the numerical time steps,  I would say.